packages/compiler-sfc/src/compileScript.ts

```diff
@@ -146,6 +155,13 @@ export function compileScript(
     !!options.refTransform
   const enablePropsTransform =
     !!options.reactivityTransform || !!options.propsDestructureTransform
+  const reactivityTransformer = enableReactivityTransform
+    ? createReactivityTransformer(
+        isObject(options.reactivityTransform)
+          ? options.reactivityTransform
+          : undefined
+      )
+    : undefined
```

Comment:
change 6

use the transformer instead of top-level apis, and allows `reactivityTransform` to be an option.

Reply:
I don't think the `createReactivityTransformer` implementation is necessary. I think the whole feature can simply be implemented by adding an extra argument to `transformAST`, (and as an extra option to the `transform` options object).